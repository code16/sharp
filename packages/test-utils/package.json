{
    "name": "@sharp/test-utils",
    "private": true,
    "main": "src/index.js",
    "version": "1.0.0"
}