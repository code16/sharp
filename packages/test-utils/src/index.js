export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC, unmockSFC } from './mockSFC';
export { mockProperty, unmockProperty, setter } from './mock-utils';
export { nextRequestFulfilled } from './moxios-utils';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}