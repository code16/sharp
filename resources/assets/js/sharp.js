import './polyfill';
import Vue from 'vue';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Trix from 'trix';
import { install as VueGoogleMaps } from 'vue2-google-maps';
import Notifications from 'vue-notification';
import locale from 'element-ui/lib/locale';
import { elLang } from './util/element-ui';


import SharpCommands from 'sharp-commands';
import SharpDashboard from 'sharp-dashboard';
import SharpEntityList from 'sharp-entity-list';
import SharpFilters from 'sharp-filters';
import SharpForm from 'sharp-form';
import SharpShow from 'sharp-show';
import SharpUI from 'sharp-ui';

import ActionView from './components/ActionView';
import LeftNav from './components/LeftNav';
import {
    ItemVisual,
    CollapsibleItem,
    NavItem,
} from 'sharp-ui';

import { store as getStore } from './store/store';
import { router as getRouter } from "./router";

locale.use(elLang());

Vue.use(Notifications);
Vue.use(VueGoogleMaps, {
    installComponents: false
});

Vue.use(VueRouter);
Vue.use(Vuex);

const router = getRouter();
const store = getStore();

Vue.use(SharpCommands, { store, router });
Vue.use(SharpDashboard, { store, router });
Vue.use(SharpEntityList, { store, router });
Vue.use(SharpFilters, { store, router });
Vue.use(SharpForm, { store, router });
Vue.use(SharpShow, { store, router });
Vue.use(SharpUI, { store, router });

window.Trix = Trix;
Vue.config.ignoredElements = [/^trix-/];


new Vue({
    el: "#sharp-app",

    components: {
        'sharp-action-view': ActionView,
        'sharp-left-nav': LeftNav,
        'sharp-collapsible-item': CollapsibleItem,
        'sharp-nav-item': NavItem,
        'sharp-item-visual': ItemVisual
    },

    created() {
        this.$on('setClass',(className,active)=> {
            this.$el.classList[active ? 'add' : 'remove'](className);
        });
        if(this.$route.query['x-access-from']) {
            this.$router.replace({
                ...this.$route,
                query: {
                    ...this.$route.query,
                    'x-access-from': undefined
                }
            });
        }
    },

    store,
    router,
});




