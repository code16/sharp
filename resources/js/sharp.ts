import '../css/vendors.css';
import '../css/app.css';

import './polyfills';
import { createApp, DefineComponent, h, nextTick } from 'vue';
import { createInertiaApp } from '@inertiajs/vue3';
import { resolvePageComponent } from 'laravel-vite-plugin/inertia-helpers';
import { progressDelay } from "@/api/progress";
import lucideIconsDefaultAttributes from 'lucide-vue-next/dist/esm/defaultAttributes';

import Form from "@/form/components/Form.vue";
import FormField from "@/form/components/Field.vue";
import ShowField from "@/show/components/Field.vue";
import { initRouter } from "@/router";

lucideIconsDefaultAttributes['aria-hidden'] = 'true'

initRouter();

createInertiaApp({
    resolve: name => resolvePageComponent(`./Pages/${name}.vue`, import.meta.glob<DefineComponent>('./Pages/**/*.vue')),
    setup({ el, App, props, plugin }) {
        const app = createApp({
            render: () => h(App, props),
        })
            .use(plugin);

        app.config.errorHandler = e => console.error(e);
        app.config.globalProperties.window = window;
        app.config.globalProperties.document = document;
        app.config.globalProperties.console = console;

        // declaring those globally because of circular dependencies
        app.component('SharpForm', Form);
        app.component('SharpFormField', FormField);
        app.component('SharpShowField', ShowField);

        app.mount(el);

        nextTick(() => {
            window.dispatchEvent(new CustomEvent('sharp:mounted'));
        });
    },
    progress: {
        delay: progressDelay,
        color: 'var(--nprogress-color, oklch(var(--primary-oklch)))',
    }
});




