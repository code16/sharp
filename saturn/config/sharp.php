<?php

return [

    "name" => "Saturn",

//    "custom_url_segment" => "admin",
    "display_sharp_version_in_title" => true,

    "extensions" => [
        "assets" => [
            "strategy" => "asset",
            "head" => [
                "/css/inject.css",
            ],
        ],

        "activate_custom_fields" => env("SHARP_CUSTOM_FIELDS", true),
    ],

    "entities" => [
        "spaceship" => [
            "list" => \App\Sharp\SpaceshipSharpList::class,
            "show" => \App\Sharp\SpaceshipSharpShow::class,
            "form" => \App\Sharp\SpaceshipSharpForm::class,
            "validator" => \App\Sharp\SpaceshipSharpValidator::class,
            "policy" => \App\Sharp\Policies\SpaceshipPolicy::class,
        ],
        "pilot" => [
            "list" => \App\Sharp\PilotSharpList::class,
            "forms" => [
                "junior" => [
                    "icon" => "far fa-user",
                    "label" => "Junior Pilot",
                    "form" => \App\Sharp\PilotJuniorSharpForm::class,
                    "validator" => \App\Sharp\PilotJuniorSharpValidator::class,
                ],
                "senior" => [
                    "icon" => "fas fa-user",
                    "label" => "Senior Pilot",
                    "form" => \App\Sharp\PilotSeniorSharpForm::class,
                    "validator" => \App\Sharp\PilotSeniorSharpValidator::class,
                ]
            ],
        ],
        "spaceship_pilot" => [
            "list" => \App\Sharp\EmbeddedEntityLists\SpaceshipPilotSharpList::class,
            "forms" => [
                "junior" => [
                    "icon" => "far fa-user",
                    "label" => "Junior Pilot",
                    "form" => \App\Sharp\PilotJuniorSharpForm::class,
                    "validator" => \App\Sharp\PilotJuniorSharpValidator::class,
                ],
                "senior" => [
                    "icon" => "fas fa-user",
                    "label" => "Senior Pilot",
                    "form" => \App\Sharp\PilotSeniorSharpForm::class,
                    "validator" => \App\Sharp\PilotSeniorSharpValidator::class,
                ]
            ],
        ],
        "passenger" => [
            "list" => \App\Sharp\PassengerSharpList::class,
            "form" => \App\Sharp\PassengerSharpForm::class,
            "validator" => \App\Sharp\PassengerSharpValidator::class,
        ],
        "travel" => [
            "list" => \App\Sharp\TravelSharpList::class,
            "form" => \App\Sharp\TravelSharpForm::class,
            "validator" => \App\Sharp\TravelSharpValidator::class,
        ],
        "user" => [
            "list" => \App\Sharp\UserSharpList::class,
            "policy" => \App\Sharp\Policies\UserPolicy::class,
            "authorizations" => [
                "delete" => false,
                "create" => false,
                "update" => false,
                "view" => false
            ]
        ],
        "account" => [
            "show" => \App\Sharp\AccountSharpShow::class,
            "form" => \App\Sharp\AccountSharpForm::class,
        ],
        "feature" => [
            "list" => \App\Sharp\FeatureSharpList::class,
            "form" => \App\Sharp\FeatureSharpForm::class,
            "validator" => \App\Sharp\FeatureSharpValidator::class,
        ],
        "test" => [
            "form" => \App\Sharp\TestForm\TestForm::class,
        ],
    ],

    "dashboards" => [
        "company_dashboard" => [
            "view" => \App\Sharp\CompanyDashboard::class,
            "policy" => \App\Sharp\Policies\CompanyDashboardPolicy::class,
        ],
        "travels_dashboard" => [
            "view" => \App\Sharp\TravelsDashboard::class,
        ],
    ],

    "global_filters" => [
        "corporation" => \App\Sharp\Filters\CorporationGlobalFilter::class
    ],

    "menu" => [
        [
            "label" => "Company",
            "entities" => [
                [
                    "label" => "Dashboard",
                    "icon" => "fas fa-tachometer-alt",
                    "dashboard" => "company_dashboard"
                ],
                [
                    "label" => "Spaceships",
                    "icon" => "fas fa-space-shuttle",
                    "entity" => "spaceship"
                ],
                [
                    "label" => "Pilots",
                    "icon" => "fas fa-user",
                    "entity" => "pilot"
                ]
            ]
        ], [
            "label" => "Travels",
            "entities" => [
                [
                    "label" => "Dashboard",
                    "icon" => "fas fa-tachometer-alt",
                    "dashboard" => "travels_dashboard"
                ],
                [
                    "label" => "Passengers",
                    "icon" => "fas fa-bed",
                    "entity" => "passenger"
                ],
                [
                    "label" => "Travel",
                    "icon" => "fas fa-suitcase",
                    "entity" => "travel"
                ],
                [
                    "label" => "Some external link",
                    "icon" => "fas fa-globe",
                    "url" => "https://google.com"
                ],
            ]
        ], [
            "label" => "Admin",
            "entities" => [
                [
                    "label" => "My account",
                    "icon" => "fas fa-user",
                    "entity" => "account",
                    "single" => true
                ],
                [
                    "label" => "Sharp users",
                    "icon" => "fas fa-user-secret",
                    "entity" => "user"
                ]
            ]
        ],
        [
            "label" => "Public website",
            "icon" => "fas fa-globe",
            "url" => "https://google.com"
        ],
        [
            "label" => "Some powerful Features",
            "icon" => "fab fa-superpowers",
            "entity" => "feature"
        ]
    ],

    "uploads" => [
        "tmp_dir" => env("SHARP_UPLOADS_TMP_DIR", "tmp"),
        "thumbnails_dir" => env("SHARP_UPLOADS_THUMBS_DIR", "thumbnails"),
    ],

    "auth" => [
//        "guard" => "sharp",
        "check_handler" => \App\Sharp\Auth\SharpCheckHandler::class,
        "login_attribute" => "email",
        "password_attribute" => "password",
        "display_attribute" => "name",
    ],

    "login_page_message_blade_path" => env("SHARP_LOGIN_PAGE_MESSAGE_BLADE_PATH", "sharp/_login-page-message"),
];
