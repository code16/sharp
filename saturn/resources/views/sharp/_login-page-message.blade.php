
<div role="alert" class="SharpNotification SharpNotification--info">
    <div class="SharpNotification__details">
        <div class="SharpNotification__text-wrapper">
            <div class="SharpNotification__title mb-3">
                Use these accounts to login:
            </div>
            <div class="SharpNotification__subtitle">
                <ul>
                    <li class="mb-2">
                        <input
                            class="SharpButton SharpButton--sm SharpButton--secondary"
                            type="button"
                            onclick="this.form.elements.login.value='admin@example.com';this.form.elements.password.value='secret'"
                            value="Admin account"
                        >
                    </li>
                    <li>
                        <input
                            class="SharpButton SharpButton--sm SharpButton--secondary"
                            type="button"
                            onclick="this.form.elements.login.value='boss@example.com';this.form.elements.password.value='secret'"
                            value="Boss account"
                        >
                        <span style="font-style: italic">(has a few more permissions)</span>
                    </li>
                </ul>
            </div>
        </div>
    </div>
</div>
