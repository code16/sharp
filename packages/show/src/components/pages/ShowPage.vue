<template>
    <div class="ShowPage">
        <div class="container">
            <template v-if="ready">
                <ActionBarShow
                    :commands="authorizedCommands"
                    :state="instanceState"
                    :state-values="stateValues"
                    :form-url="formUrl"
                    :back-url="backUrl"
                    :can-edit="canEdit"
                    :can-change-state="canChangeState"
                    :show-back-button="showBackButton"
                    @command="handleCommandRequested"
                    @state-change="handleStateChanged"
                />

                <template v-for="section in layout.sections">
                    <div class="ShowPage__section" :class="sectionClasses(section)" v-show="isSectionVisible(section)">
                        <template v-if="section.title">
                            <h2 class="ShowPage__section-title mb-2">{{ section.title }}</h2>
                        </template>
                        <div class="ShowPage__section-content">
                            <Grid class="ShowPage__section-grid"
                                :rows="[section.columns]"
                                :col-class="sectionColClass"
                                v-slot="{ itemLayout:column }"
                            >
                                <Grid class="ShowPage__fields-grid"
                                    :rows="column.fields"
                                    :row-class="fieldsRowClass"
                                    v-slot="{ itemLayout:fieldLayout }"
                                >
                                    <template v-if="fieldOptions(fieldLayout)">
                                        <ShowField
                                            :options="fieldOptions(fieldLayout)"
                                            :value="fieldValue(fieldLayout)"
                                            :config-identifier="fieldLayout.key"
                                            :layout="fieldLayout"
                                            @visible-change="handleFieldVisibilityChanged(fieldLayout.key, $event)"
                                        />
                                    </template>
                                    <template v-else>
                                        <UnknownField :name="fieldLayout.key" />
                                    </template>
                                </Grid>
                            </Grid>
                        </div>
                    </div>
                </template>

            </template>
        </div>

        <CommandFormModal :form="commandCurrentForm" ref="commandForm" />
        <CommandViewPanel :content="commandViewContent" @close="handleCommandViewPanelClosed" />
    </div>
</template>

<script>
    import { mapGetters } from 'vuex';
    import { formUrl, getBackUrl, lang, showAlert } from 'sharp';
    import { CommandFormModal, CommandViewPanel } from 'sharp-commands';
    import { Grid } from 'sharp-ui';
    import { UnknownField } from 'sharp/components';
    import { withCommands } from 'sharp/mixins';
    import ActionBarShow from "../ActionBar";

    import ShowField from '../Field';

    export default {
        mixins: [withCommands],

        components: {
            ActionBarShow,
            Grid,
            ShowField,
            UnknownField,
            CommandFormModal,
            CommandViewPanel,
        },

        data() {
            return {
                ready: false,
                fieldsVisible: null,
            }
        },

        computed: {
            ...mapGetters('show', [
                'entityKey',
                'instanceId',
                'fields',
                'layout',
                'data',
                'config',
                'breadcrumb',
                'instanceState',
                'canEdit',
                'authorizedCommands',
                'stateValues',
                'canChangeState',
            ]),

            formUrl() {
                return formUrl({
                    entityKey: this.entityKey,
                    instanceId: this.instanceId,
                });
            },
            backUrl() {
                return getBackUrl(this.breadcrumb);
            },
            showBackButton() {
                return !!this.backUrl;
            },
        },

        methods: {
            fieldOptions(layout) {
                const options = this.fields
                    ? this.fields[layout.key]
                    : null;
                if(!options) {
                    console.error(`Show page: unknown field "${layout.key}"`);
                }
                return options;
            },
            fieldValue(layout) {
                return this.data
                    ? this.data[layout.key]
                    : null;
            },
            isFieldVisible(layout) {
                return !this.fieldsVisible || this.fieldsVisible[layout.key] !== false;
            },
            fieldsRowClass(row) {
                const fieldsTypeClasses = row.map(fieldLayout => {
                    const field = this.fieldOptions(fieldLayout);
                    return `ShowPage__fields-row--${field.type}`;
                });
                return [
                    'ShowPage__fields-row',
                    ...fieldsTypeClasses,
                ]
            },
            sectionClasses(section) {
                return {
                    'ShowPage__section--no-container': this.sectionHasField(section, 'entityList'),
                }
            },
            sectionColClass() {
                return 'ShowPage__section-col';
            },
            sectionFields(section) {
                return section.columns.reduce((res, column) => [...res, ...column.fields.flat()], []);
            },
            isSectionVisible(section) {
                const sectionFields = this.sectionFields(section);
                return sectionFields.some(fieldLayout => this.isFieldVisible(fieldLayout));
            },
            sectionHasField(section, type) {
                const sectionFields = this.sectionFields(section);
                return sectionFields.some(fieldLayout => {
                    const options = this.fieldOptions(fieldLayout);
                    return options && options.type === type;
                });
            },
            handleFieldVisibilityChanged(key, visible) {
                this.fieldsVisible = {
                    ...this.fieldsVisible,
                    [key]: visible,
                }
            },
            handleCommandRequested(command) {
                this.sendCommand(command, {
                    postCommand: () => this.$store.dispatch('show/postCommand', { command }),
                    postForm: data => this.$store.dispatch('show/postCommand', { command, data }),
                    getFormData: () => this.$store.dispatch('show/getCommandFormData', { command }),
                });
            },
            handleStateChanged(state) {
                return this.$store.dispatch('show/postState', state)
                    .then(data => {
                        this.handleCommandActionRequested(data.action, data);
                    })
                    .catch(error => {
                        const data = error.response.data;
                        if(error.response.status === 422) {
                            showAlert(data.message, {
                                title: lang('modals.state.422.title'),
                                isError: true,
                            });
                        }
                    });
            },
            handleRefreshCommand() {
                this.init();
            },
            initCommands() {
                this.addCommandActionHandlers({
                    'refresh': this.handleRefreshCommand,
                });
            },
            async init() {
                await this.$store.dispatch('show/setEntityKey', this.$route.params.entityKey);
                await this.$store.dispatch('show/setInstanceId', this.$route.params.instanceId);
                await this.$store.dispatch('show/get');

                this.ready = true;
            }
        },

        beforeMount() {
            this.init();
            this.initCommands();
        },
    }
</script>