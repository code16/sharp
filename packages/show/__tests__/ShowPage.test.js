import merge from 'lodash/merge';
import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import ShowPage from "../src/components/pages/ShowPage";
import showModule from "../src/store/show";
import { formUrl } from "sharp";
import { MockInjections } from "@sharp/test-utils";

jest.mock('../src/store/show');
jest.mock('sharp/consts', () => ({
    BASE_URL: 'BASE_URL'
}));
jest.mock('sharp');


describe('show page', () => {

    function createWrapper({ storeModule, ...options } = {}) {
        const localVue = createLocalVue();
        localVue.use(Vuex);

        const wrapper = shallowMount(ShowPage, {
            provide: MockInjections.provide,
            extends: {
                computed: {
                    breadcrumb: () => [],
                    config: () => ({}),
                },
            },
            created() {
                jest.spyOn(this, 'init').mockImplementation();
                jest.spyOn(this.$store, 'dispatch').mockImplementation(()=>Promise.resolve());
            },
            store: new Vuex.Store({
                modules: {
                    'show': merge(showModule, storeModule),
                }
            }),
            // language=Vue
            stubs: {
                Grid:
                    `<div class="MOCKED_SharpGrid">
                        <slot :item-layout="rows[0][0]" />
                    </div>`,
            },
            mocks: {
                $route: { params: { } },
            },
            localVue,
            ...options,
        });
        return wrapper;
    }

    test('can mount', () => {
        expect(createWrapper().html()).toMatchSnapshot();
    });

    test('can mount with fields', () => {
        const wrapper = createWrapper({
            computed: {
                layout: () => ({
                    sections: [
                        {
                            title: 'Section title',
                            columns: [{
                                fields: [[{ key:'name' }]]
                            }]
                        }
                    ]
                }),
                formUrl: () => 'formUrl',
            },
        });

        wrapper.setMethods({
            fieldOptions: () => ({}),
            fieldValue: () => ({}),
        });
        wrapper.setData({ ready: true });
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('can mount with unknown field', () => {
        const wrapper = createWrapper({
            computed: {
                layout: () => ({
                    sections: [
                        {
                            title: 'Section title',
                            columns: [{
                                fields: [[{ key:'name' }]]
                            }]
                        }
                    ]
                }),
                formUrl: () => 'formUrl',
            },
        });

        wrapper.setMethods({
            fieldOptions: () => null,
        });
        wrapper.setData({ ready: true });
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('can mount with no container section', () => {
        const wrapper = createWrapper({
            computed: {
                layout: () => ({
                    sections: [
                        {
                            title: 'Section title',
                            columns: [{
                                fields:[[{ key:'name' }]]
                            }]
                        }
                    ]
                }),
                formUrl: () => 'formUrl',
            },
        });
        wrapper.setMethods({
            fieldOptions: () => ({ type:'entityList' }),
        });
        wrapper.setData({ ready: true });
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('formUrl', () => {
        formUrl.mockReturnValue('formUrl');
        const wrapper = createWrapper({
            storeModule: {
                getters: {
                    entityKey: () => 'entityKey',
                    instanceId: () => 'instanceId',
                }
            }
        });
        expect(wrapper.vm.formUrl).toEqual('formUrl');
        expect(formUrl).toHaveBeenCalledWith({
            entityKey: 'entityKey',
            instanceId: 'instanceId',
        });
    });

    test('fieldOptions', () => {
        let wrapper;
        const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation();

        wrapper = createWrapper({
            computed: {
                fields: () => ({
                    name: 'options'
                })
            },
        });
        expect(wrapper.vm.fieldOptions({ key:'name' })).toBe('options');

        wrapper = createWrapper({
            computed: {
                fields: () => ({}),
            }
        });
        consoleErrorSpy.mockClear();
        expect(wrapper.vm.fieldOptions('name')).toBeUndefined();
        expect(console.error).toHaveBeenCalled();

        wrapper = createWrapper({
            computed: {
                fields: () => null,
            }
        });
        consoleErrorSpy.mockClear();
        expect(wrapper.vm.fieldOptions('name')).toBe(null);
        expect(console.error).toHaveBeenCalled();
    });

    test('fieldValue', () => {
        let wrapper;

        wrapper = createWrapper({
            computed: {
                data: () => ({
                    name: 'value'
                })
            },
        });
        expect(wrapper.vm.fieldValue({ key:'name' })).toBe('value');

        wrapper = createWrapper({
            computed: {
                data: () => ({})
            }
        });
        expect(wrapper.vm.fieldValue({ key:'name' })).toBeUndefined();
        expect(console.error).toHaveBeenCalled();

        wrapper = createWrapper({
            computed: {
                fields: () => null,
            }
        });
        expect(wrapper.vm.fieldValue({ key:'name' })).toBe(null);
        expect(console.error).toHaveBeenCalled();
    });

    test('change state + command action integration', async () => {
        const wrapper = createWrapper();

        wrapper.vm.init.mockClear();
        wrapper.vm.$store.dispatch.mockReturnValueOnce(Promise.resolve({
            action: 'refresh',
        }));
        await wrapper.vm.handleStateChanged({});
        expect(wrapper.vm.init).toHaveBeenCalled();
    });
});