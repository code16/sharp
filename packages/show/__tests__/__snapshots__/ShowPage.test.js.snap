// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show page can mount 1`] = `
<div class="ShowPage">
	<div class="container">
		<!---->
	</div>
	<commandformmodal-stub></commandformmodal-stub>
	<commandviewpanel-stub></commandviewpanel-stub>
</div>
`;

exports[`show page can mount with fields 1`] = `
<div class="ShowPage">
	<div class="container">
		<actionbarshow-stub formurl="formUrl"></actionbarshow-stub>
		<div class="ShowPage__section">
			<h2 class="ShowPage__section-title mb-2">Section title</h2>
			<div class="ShowPage__section-content">
				<div class="MOCKED_SharpGrid ShowPage__section-grid">
					<div class="MOCKED_SharpGrid ShowPage__fields-grid">
						<showfield-stub configidentifier="name" value="[object Object]" options="[object Object]" layout="[object Object]"></showfield-stub>
					</div>
				</div>
			</div>
		</div>
	</div>
	<commandformmodal-stub></commandformmodal-stub>
	<commandviewpanel-stub></commandviewpanel-stub>
</div>
`;

exports[`show page can mount with no container section 1`] = `
<div class="ShowPage">
	<div class="container">
		<actionbarshow-stub formurl="formUrl"></actionbarshow-stub>
		<div class="ShowPage__section ShowPage__section--no-container">
			<h2 class="ShowPage__section-title mb-2">Section title</h2>
			<div class="ShowPage__section-content">
				<div class="MOCKED_SharpGrid ShowPage__section-grid">
					<div class="MOCKED_SharpGrid ShowPage__fields-grid">
						<showfield-stub configidentifier="name" options="[object Object]" layout="[object Object]"></showfield-stub>
					</div>
				</div>
			</div>
		</div>
	</div>
	<commandformmodal-stub></commandformmodal-stub>
	<commandviewpanel-stub></commandviewpanel-stub>
</div>
`;

exports[`show page can mount with unknown field 1`] = `
<div class="ShowPage">
	<div class="container">
		<actionbarshow-stub formurl="formUrl"></actionbarshow-stub>
		<div class="ShowPage__section">
			<h2 class="ShowPage__section-title mb-2">Section title</h2>
			<div class="ShowPage__section-content">
				<div class="MOCKED_SharpGrid ShowPage__section-grid">
					<div class="MOCKED_SharpGrid ShowPage__fields-grid">
						<unknownfield-stub name="name"></unknownfield-stub>
					</div>
				</div>
			</div>
		</div>
	</div>
	<commandformmodal-stub></commandformmodal-stub>
	<commandviewpanel-stub></commandviewpanel-stub>
</div>
`;
